tests/spec_tests/query_tests/distinct.yml

```
@@ -0,0 +1,132 @@
+catalog_data:
+  db:
+    foo:
+      - { _id: 1, a: 1, b: 2 }
+      - { _id: 2, a: 1, b: 2 }
+      - { _id: 3, a: 2, b: 2 }
+      - { _id: 4, b: 2, a: 1 }
+      - { _id: 5, a: 2, b: 2, c: true, d: "a", e: null }
+      - { _id: 6, a: 2, b: 2, c: true, d: "a", e: null }
+
+    documents:
+      - { _id: 1, doc: { a: 1, b: 2 } }
+      - { _id: 2, doc: { a: 1, b: 2 } }
+      - { _id: 3, doc: { a: 2, b: 2 } }
+      - { _id: 4, doc: { a: 2, b: 3 } }
+      - { _id: 5, doc: { b: 2, a: 1 } }
+
+    arrays:
+      - { _id: 1, arr: [1, 2, 3] }
+      - { _id: 2, arr: [1, 2, 3] }
+      - { _id: 3, arr: [2, 3, 4] }
+      - { _id: 4, arr: [3, 2, 1] }
+
+catalog_schema:
+  {
+    "db": {
+      "foo": {
+        "bsonType": "object",
+        "required": ["_id", "a", "b"],
+        "additionalProperties": false,
+        "properties": {
+          "_id":   { "bsonType": "int" },
+          "a":     { "bsonType": "int" },
+          "b":     { "bsonType": "int" },
+          "c": {
+            "anyOf": [
+              { "bsonType": "bool" },
+              { "bsonType": "null" }
+            ]
+          },
+          "d": {
+            "anyOf": [
+              { "bsonType": "string" },
+              { "bsonType": "null" }
+            ]
+          },
+          "e": {
+            "bsonType": "null"
+          }
+        }
+      },
+      "documents": {
+        "bsonType": "object",
+        "required": ["_id", "doc"],
+        "additionalProperties": false,
+        "properties": {
+          "_id": { "bsonType": "int" },
+          "doc": {
+            "bsonType": "object",
+            "required": ["a", "b"],
+            "properties": {
+              "a": { "bsonType": "int" },
+              "b": { "bsonType": "int" }
+            }
+          }
+        }
+      },
+      "arrays": {
+        "bsonType": "object",
+        "required": ["_id", "arr"],
+        "additionalProperties": false,
+        "properties": {
+          "_id": { "bsonType": "int" },
+          "arr": {
+            "bsonType": "array",
+            "items": { "bsonType": "int" }
+          }
+        }
+      }
+    }
+  }
+
+tests:
+  - description: "SELECT DISTINCT * from distinct ensures unique rows for entire document"
```

```suggestion
  - description: "SELECT DISTINCT * ensures unique rows for entire document"
```
What does "from distinct" mean here? I think the test description works when we drop those words, but open to adding them back in some way if you had something specific in mind.

---

Oh, I had the collection originally named `distinct` but even that's unnecessary to have here, I'll update it.